projects/archiver/src/notifications/device-notifications.ts

```diff
@@ -1,30 +1,59 @@
-import { IssuePublicationIdentifier, Issue } from '../../common'
+import uuid from 'uuidv4'
+import { attempt } from '../../../backend/utils/try'
+import fetch from 'node-fetch'
+import dateFormat from 'dateformat'
 
-interface Topic {
-    type: string
+type EditionNotificationTypes = 'editions'
+
+type NotificationSenders = 'editions-backend'
+
+interface EditionTopic {
+    type: EditionNotificationTypes
     name: string
 }
 
 interface ScheduleAppStoreNotificationPayload {
     id: string
-    type: string
-    topic: Topic[]
+    type: EditionNotificationTypes
+    topic: EditionTopic[]
     key: string
     name: string
     date: string
-    sender: string
+    sender: NotificationSenders
+}
+
+const createDatePathParam = (now: Date) => {
+    const THREE_AT_NIGHT = 'T03:00:00Z'
+    return `${dateFormat(now, 'yyyy-mm-dd')}${THREE_AT_NIGHT}`
+}
+
+const createScheduleNotificationEndpoint = (domain: string, now: Date) => {
+    const sheduleDate = createDatePathParam(now)
+    return `${domain}/push/schedule/${sheduleDate}`
+}
+
+export interface IssueNotifiactionData {
+    key: string
+    name: string
+    issueDate: string
 }
 
-export const scheduleAppStoreNotification = (
-    issue: Issue,
-    issuePublication: IssuePublicationIdentifier,
-): any => {
-    console.log('scheduling App Store Notification')
-    const { issueDate } = issuePublication
-    const { key, name } = issue
+export const scheduleAppStoreNotification = async (
+    issueData: IssueNotifiactionData,
+    cfg: { domain: string; apiKey: string },
+    now: Date,
+) => {
+    const { domain, apiKey } = cfg
+    const reqEndpoint = createScheduleNotificationEndpoint(domain, now)
+    console.log(
+        'sending schedule App Store Notification Request to',
+        reqEndpoint,
+    )
+
+    const { key, name, issueDate } = issueData
 
-    const issueAppStoreNotification: ScheduleAppStoreNotificationPayload = {
-        id: '123',
+    const payload: ScheduleAppStoreNotificationPayload = {
+        id: uuid.fromString(key),
         type: 'editions',
         topic: [{ type: 'editions', name: 'uk' }],
```

Comment:
As above, we should have a TODO here for when we try to make this generic.

Reply:
done